satpy/readers/yaml_reader.py

```diff
@@ -322,22 +331,25 @@ def check_file_covers_area(file_handler, check_area):
         return True
 
     def find_required_filehandlers(self, requirements, filename_info):
-        """Find the necessary fhs for the current filehandler.
+        """Find the necessary file handlers for the given requirements.
 
         We assume here requirements are available.
+
+        Raises:
+            KeyError, if no handler for the given requirements is available
+            RumtimeError, if there is a handler for the given requirements,
+            but it doesn't match the filename info
         """
         req_fh = []
+        filename_info = set(filename_info.items())
         if requirements:
             for requirement in requirements:
                 for fhd in self.file_handlers[requirement]:
-                    # FIXME: Isn't this super wasteful? filename_info.items()
-                    # every iteration?
-                    if (all(item in filename_info.items()
-                            for item in fhd.filename_info.items())):
+                    if set(fhd.filename_info.items()).issubset(filename_info):
                         req_fh.append(fhd)
                         break
                 else:
-                    raise RuntimeError('No matching file in ' + requirement)
+                    raise RuntimeError(requirement)
```

Comment:
Even if this gets caught and a warning issued it would be nice to have a proper error message. Maybe `"No matching requirement file of type: {}".format(requirement)`?

Reply:
Good point. I use the exception to pass the name of the missing requirement. If I catch ``KeyError`` and ``RuntimeError`` separately, I can use the ``RuntimeError`` message as a warning.